src/algorand_transactions/transactions_group.rs

```diff
@@ -0,0 +1,240 @@
+    AlgorandSignedTransaction,
+    AlgorandTransaction,
+};
+
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct TxGroup {
+    #[serde(rename = "txlist", default)]
+    tx_group_hashes: Vec<AlgorandHash>,
+
+    #[serde(skip_serializing)]
+    txs: Vec<AlgorandTransaction>,
+
+    #[serde(skip_serializing)]
+    group_id: Option<AlgorandHash>,
+}
+
+impl TxGroup {
+    const MAX_TX_GROUP_SIZE: usize = 16;
+
+    pub fn new(transactions: &[AlgorandTransaction]) -> Result<TxGroup> {
+        let a = transactions.iter().map(|tx| tx.to_raw_tx_id()).collect();
+        match a {
+            Ok(res) => Ok(Self {
+                txs: transactions.to_vec(),
+                group_id: None,
+                tx_group_hashes: res,
+            }),
+            _ => Err(AlgorandError::Custom("error".to_string())),
+        }
+    }
+
+    pub fn assign_group_id(&mut self) -> Result<AlgorandHash> {
+        let group_id = TxGroup::compute_group_id(self)?;
+        self.txs = self
+            .txs
+            .iter()
+            .map(|tx| tx.assign_group_id(group_id))
+            .collect();
+        self.group_id = Some(group_id);
+        Ok(group_id)
+    }
+
+    pub fn compute_group_id(&self) -> Result<AlgorandHash> {
+        if self.tx_group_hashes.is_empty() {
+            return Err(AlgorandError::Custom("Empty transactions list".to_string()));
+        }
+        if self.tx_group_hashes.len() > Self::MAX_TX_GROUP_SIZE {
+            return Err(AlgorandError::Custom(
+                "Too  many transactions in group!".to_string(),
+            ));
+        }
+        let hashed = sha2::Sha512_256::digest(self.encode_for_hashing()?);
+        AlgorandHash::from_slice(hashed.as_slice())
+    }
+
+    fn to_msg_pack_bytes(&self) -> Result<Bytes> {
+        Ok(rmp_serde::to_vec_named(&self)?)
+    }
+
+    fn prefix_group_byte(bytes: &[Byte]) -> Bytes {
+        let suffix = bytes;
+        let mut prefix = b"TG".to_vec();
+        prefix.extend_from_slice(suffix);
+        prefix
+    }
+
+    pub fn encode_for_hashing(&self) -> Result<Bytes> {
+        self.to_msg_pack_bytes()
+            .map(|ref msg_pack_bytes| Self::prefix_group_byte(msg_pack_bytes))
+    }
+
+    pub fn to_hex(txns: Vec<AlgorandSignedTransaction>) -> Result<String> {
+        Ok(txns
+            .iter()
+            .map(|t| t.to_hex())
+            .collect::<Result<Vec<String>>>()?
+            .join(""))
+    }
+
+    pub fn sign_transactions(&self, keys: &[&AlgorandKeys]) -> Result<String> {
+        let signed_txs: Result<Vec<AlgorandSignedTransaction>> = match keys.len() {
+            len if len == 1 => self.txs.iter().map(|tx| tx.sign(keys[0])).collect(),
+            len if len == self.txs.len() => self
+                .txs
+                .iter()
+                .enumerate()
+                .map(|(i, tx)| tx.sign(keys[i]))
+                .collect(),
+            _ => Err(AlgorandError::Custom("error1".to_string())),
```

Comment:
Another usage of `AlgorandError::Custom`

Reply:
See my review (once done).